src/enclave.rs

```diff
@@ -0,0 +1,227 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Enclave object
+
+extern crate std;
+
+use crate::page::{Class, Flags, SecInfo, Secs};
+use crate::signature::Signature;
+
+use std::fs::{File, OpenOptions};
+use std::io::Result;
+use std::marker::PhantomData;
+
+use iocuddle::{Group, Ioctl, Write, WriteRead};
+
+const SGX_IOCTL: u8 = 0xA4;
+
+/// Wraps an enclave file descriptor.
+pub struct Enclave(File);
+
+impl From<Enclave> for File {
+    fn from(enclave: Enclave) -> File {
+        enclave.0
+    }
+}
+
+impl Enclave {
+    /// The path of the device file.
+    pub const DEVICE: &'static str = "/dev/sgx_enclave";
+
+    /// Create a new instance.
+    pub fn new() -> Result<Self> {
+        OpenOptions::new()
+            .read(true)
+            .write(true)
+            .open(Self::DEVICE)
+            .map(Self)
+    }
+
+    pub fn try_clone(&mut self) -> Result<Self> {
+        Ok(Self(self.0.try_clone()?))
+    }
+
+    /// ENCLS[ECREATE] wrapper
+    pub fn create(&mut self, secs: &Secs) -> Result<()> {
+        const CREATE: Ioctl<Write, &'static Create<'static>> =
```

Comment:
All of the `'static` lifetimes in all of these consts shouldn't be required, Rust should be smart enough to know that `'static` is the only valid lifetime for a static and it should infer that appropriately.

Reply:
Right, I fixed this now, thank you.